include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -236,56 +296,119 @@ struct GemmKernel
-                    sequence<false, GemmPipeline::kPadN>{});
-            }
-            else
+            auto c_pad_view = [&]() {
+                if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
+                {
+                    return pad_tensor_view(
+                        c_tensor_view,
+                        make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                        sequence<false, GemmPipeline::kPadN>{});
+                }
+                else
+                {
+                    return pad_tensor_view(
+                        c_tensor_view,
+                        make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                        sequence<GemmPipeline::kPadM, false>{});
+                }
+            }();
+            auto CBlockWindow_pad = make_tile_window(
+                c_pad_view,
+                make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                {i_m, i_n});
+            EpiloguePipeline{}(CBlockWindow_pad, c_block_tile);
+        }
+        else
+        {
+            auto c_tensor_view = [&]() {
+                if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
+                {
+                    return make_naive_tensor_view<address_space_enum::global,
+                                                  memory_operation_enum::atomic_add>(
+                        c_start,
+                        make_tuple(kargs.M, kargs.N),
+                        make_tuple(kargs.stride_C, 1),
+                        number<GemmPipeline::VectorSizeC>{},
+                        number<1>{});
+                }
+                else
+                {
+                    return make_naive_tensor_view<address_space_enum::global,
+                                                  memory_operation_enum::atomic_add>(
+                        c_start,
+                        make_tuple(kargs.M, kargs.N),
+                        make_tuple(1, kargs.stride_C),
+                        number<1>{},
+                        number<1>{});
+                }
+            }();
+
+            auto c_pad_view = [&]() {
+                if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
+                {
+                    return pad_tensor_view(
+                        c_tensor_view,
+                        make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                        sequence<false, GemmPipeline::kPadN>{});
+                }
+                else
+                {
+                    return pad_tensor_view(
+                        c_tensor_view,
+                        make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                        sequence<GemmPipeline::kPadM, false>{});
+                }
+            }();
```

Comment:
Padding is the same regardless of kbatch right?

Reply:
We have check in isSupported -> `kargs.K % TilePartitioner::kK != 0 && GemmPipeline::kPadK == fals`e so it should be fine

Reply:
I rather meant that the padding code is Kbatch value agnostic.

Reply:
Yes, it is agnostic